Tests/RemoteDumpTest.php

```diff
@@ -154,43 +226,185 @@ public function responseCodes()
+    {
+        Config::set('protector.baseDirectory', __FUNCTION__);
+
+        $result = $this->runProtectedMethod('getConfigValueForKey', ['baseDirectory']);
+
+        $this->assertEquals(__FUNCTION__, $result);
+    }
+
+    /**
+     * @test
+     */
+    public function canResolveBaseDirectoryFromClosure()
+    {
+        $functionName = __FUNCTION__;
 
-        $method = $this->getAccessibleReflectionMethod('getConfiguredHttpRequest');
+        Config::set('protector.baseDirectory', fn() => $functionName);
 
+        $result = $this->runProtectedMethod('getConfigValueForKey', ['baseDirectory']);
+
+        $this->assertEquals($functionName, $result);
+    }
+
+    /**
+     * Sets the env key name for the private key.
+     * @test
+     */
+    public function validateUsersPrivateKeyName()
+    {
+        $this->protector->setPrivateKeyName(__FUNCTION__);
+        $privateKeyName = $this->protector->getPrivateKeyName();
+
+        $this->assertEquals(__FUNCTION__, $privateKeyName);
+    }
+
+    /**
+     * @test
+     */
+    public function failDecryptingOnInvalidString()
+    {
         $this->expectException(InvalidConfigurationException::class);
-        $method->invoke(app('protector'));
+        $this->protector->decryptString(base64_encode(__FUNCTION__));
     }
 
     /**
-     * @param $method
-     *
-     * @return ReflectionMethod
+     * @test
      */
-    protected function getAccessibleReflectionMethod($method): ReflectionMethod
+    public function canCreateFilename()
     {
-        $reflectionProtector = new ReflectionClass(app('protector'));
-        $method              = $reflectionProtector->getMethod($method);
+        $fileName  = $this->protector->createFilename();
+        $structure = '%s %d-%d-%d %d-%d %x.sql';
+
+        $this->assertIsString($fileName);
+        $this->assertStringMatchesFormat($structure, $fileName);
+    }
 
-        $method->setAccessible(true);
+    /**
+     * @test
+     */
+    public function canGetAuthToken()
+    {
+        $authToken = $this->runProtectedMethod('getAuthToken', []);
 
-        return $method;
+        $this->assertEquals(env('PROTECTOR_AUTH_TOKEN'), $authToken);
+    }
+
+    /**
+     * @test
+     */
+    public function setAuthToken()
+    {
+        $this->runProtectedMethod('setAuthToken', [__FUNCTION__]);
+        $authToken = $this->runProtectedMethod('getAuthToken', []);
+
+        $this->assertEquals(__FUNCTION__, $authToken);
+    }
+
+    /**
+     * @test
+     */
+    public function setAuthTokenKeyName()
+    {
+        $this->runProtectedMethod('setAuthTokenKeyName', [__FUNCTION__]);
+        $authTokenKeyName = $this->runProtectedMethod('getAuthTokenKeyName', []);
+
+        $this->assertEquals(__FUNCTION__, $authTokenKeyName);
+    }
+
+    /**
+     * @test
+     */
+    public function canGetServerUrl()
+    {
+        $this->assertEquals(config('protector.remoteEndpoint.serverUrl'), $this->protector->getServerUrl());
```

Comment:
this doesnt protect against both values being null for some reason. the test should be more robust.

Reply:
The config value of the server url is set in the setUp Method.